AEPCore/Sources/rules/LaunchRulesEngine.swift

```diff
@@ -330,10 +336,130 @@ public class LaunchRulesEngine {
         dict[LaunchRulesEngine.CONSEQUENCE_EVENT_DATA_KEY_TYPE] = consequence.type
         return parentEvent.createChainedEvent(name: LaunchRulesEngine.CONSEQUENCE_EVENT_NAME, type: EventType.rulesEngine, source: EventSource.responseContent, data: [LaunchRulesEngine.CONSEQUENCE_EVENT_DATA_KEY_CONSEQUENCE: dict])
     }
+
+    /// Process a schema consequence event. Handles different schema types including event history operations.
+    /// - Parameters:
+    ///   - consequence: the RuleConsequence which contains the schema details
+    ///   - processedEvent: the event that triggered the rule
+    ///   - data: a `Traversable` object used for token resolution
+    private func processSchemaConsequence(consequence: RuleConsequence, processedEvent: Event, data: Traversable) {
+        guard let schema = consequence.schema, let _ = consequence.data, let _ = consequence.detailID else {
+            Log.error(label: LOG_TAG, "(\(self.name)) : Unable to process Schema Consequence with ID: \(consequence.id). A Schema Consequence requires 'schema', 'data', and 'id' properties.")
+            return
+        }
+
+        switch schema {
+        case LaunchRulesEngine.CONSEQUENCE_SCHEMA_EVENT_HISTORY:
+            processEventHistoryOperation(consequence: consequence, processedEvent: processedEvent, data: data)
+        default:
+            Log.warning(label: LOG_TAG, "(\(self.name)) : Unsupported schema type: \(schema)")
+        }
+    }
+
+    /// Process an event history operation consequence. Records events in the Event History database.
+    /// - Parameters:
+    ///   - consequence: the RuleConsequence which contains the event history operation details
+    ///   - processedEvent: the event that triggered the rule
+    ///   - data: a `Traversable` object used for token resolution
+    private func processEventHistoryOperation(consequence: RuleConsequence, processedEvent: Event, data: Traversable) {
```

Comment:
let's get the magic strings out of this method: 
- `operation`
- `keys`
- `~`
- `content`